Automation/NSG/deploy-nsg.ps1

```diff
@@ -0,0 +1,104 @@
+param(
+    # Change if you want a different region limit notification
+    $ResourceLimit = 15,
+
+    # Found in Azure secrets
+    [Parameter(Mandatory=$True)]
+    [string]
+    $ServicePrincipleUsername,
+
+    # Your tenant Id in Azure
+    [Parameter(Mandatory=$True)]
+    [string]
+    $TenantId,
+
+    # Name for the resource group you will create
+    [Parameter(Mandatory=$True)]
+    [string]
+    $ResourceGroupName,
+
+    # Region. ex. westus, eastus...
+    [Parameter(Mandatory=$True)]
+    [string]
+    $Region,
+
+    [Parameter(Mandatory=$True)]
+    [string]
+    $DeploymentName,
+
+    # File path to the NSG deployment template
+    [Parameter(Mandatory=$True)]
+    [string]
+    $TemplateFilePath,
+
+    # File path to the NSG parameter template
+    [Parameter(Mandatory=$True)]
+    [string]
+    $ParameterFilePath
+)
```

Comment:
add the secret/password as a parameter as well so that doesn't need to be an extra step

Reply:
I decided to leave the secret/password as it's own step since the default prompt was more verbose about what password you are entering and what it is for.